jax/_src/lax/lax.py

```diff
@@ -402,8 +402,6 @@ def convert_element_type(operand: Array, new_dtype: DType) -> Array:
   if type(operand) in dtypes.python_scalar_dtypes:
     operand = np.asarray(operand, new_dtype)
   old_dtype = dtypes.canonicalize_dtype(_dtype(operand))
-  if old_dtype == new_dtype:
-    return operand
```

Comment:
Note: I removed this because `convert_element_type` also now removes the weak type from the input, and it is cleaner to let this happen via the common abstract eval than to create a new primitive to strip the weak type from operand.